File: scala/robot-name/src/main/scala/robot.scala
import scala.util.Random

class Robot {
    private var _name : String = newName

    def name : String = {
        _name
    }

    def reset() : Unit = {
        _name = newName
    }

    private def newName : String = {
        firstPartName + lastPartName
    }

    private def firstPartName : String = {
        Random.shuffle('A'.to('Z').toList).take(2).mkString
    }

    private def lastPartName : String = {
        Random.shuffle(100.to(999).toList).head.toString
    }

}

File: python/robot-name/robot.py
from string import ascii_uppercase, digits
from random import choice

class Robot:
    @staticmethod
    def _name_generator():
        return (choice(ascii_uppercase) +
        choice(ascii_uppercase) +
        choice(digits) +
        choice(digits) +
        choice(digits))

    def reset(self):
        self.name = self._name_generator

    def __init__(self):
        self.name = self._name_generator()